### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by Lisio** [mongodb/mongo-php-driver#514]
### Description

Driver causes memory leak for the collections with at least 9998 docs (9997 docs work fine).

### Environment

Ubuntu 16.04 x64
PHP 7.1.0
mongodb 1.2.3

```
$ php -i | grep -E 'mongodb|libmongoc|libbson'
mongodb
libbson bundled version => 1.5.3
libmongoc bundled version => 1.5.3
libmongoc SSL => enabled
libmongoc SSL library => OpenSSL
libmongoc crypto => enabled
libmongoc crypto library => libcrypto
libmongoc crypto system profile => disabled
libmongoc SASL => disabled
mongodb.debug => no value => no value
```

### Test Script

```
<?php

$mongo = new \MongoDB\Driver\Manager("mongodb://username:password@127.0.0.1:27017/dbname");
$query = new \MongoDB\Driver\Query([]);

while (true) {
    $mongo->executeQuery('dbname.test', $query)->toArray();
    echo memory_get_peak_usage() . PHP_EOL;
}
```

Debug (9997 docs): http://kostin.info/9997.zip (1.5Mb)
Debug (9998 docs): http://kostin.info/9998.zip (1.5Mb)

**Comment by Lisio** [comment]
As you can see in 9998.stdout the memory grows by 1919616 bytes each step, which is 1919616 / 192 = 9998.
For 12345 docs it also grows by 1919616 bytes.
For 23456 docs it grows by 3839616 bytes (3839616 / 192) = 19998
For 34567 docs it grows by 5759616 bytes (5759616 / 192) = 29998

Also, if I create more complex documents, the difference between each subsequent step is still the same:
```
db.test.drop(); for (var i = 0; i < 12345; ++i) {db.test.insert({a: "some text string", b: "another text string", c: "yet another text string"});}
```
```
$ php index.php
21511232
23431392
25351008
27270624
29190240
31109856
33029472
34949088
36868704
38788320
```

What's the magic with this x9998 numbers?

**Comment by jmikola** [comment]
Cross-referencing with [PHP-894](https://jira.mongodb.org/browse/PHPC-894).

**Comment by jmikola** [comment]
Pull request in #515.

**Comment by Lisio** [comment]
With PR #515 look good:
```
db.test.drop(); for (var i = 0; i < 123456; ++i) {db.test.insert({a: "some text string", b: "another text string", c: "yet another text string"});}
```
```
$ php index.php 
154468176
154468336
154468336
154468336
154468336
154468336
154468336
154468336
154468336
154468336
```

**Comment by jmikola** [comment]
Closing this, as [PHPC-894](https://jira.mongodb.org/browse/PHPC-894). was released in [1.2.4](https://github.com/mongodb/mongo-php-driver/releases/tag/1.2.4).
